optimade/adapters/structures/pymatgen.py

```diff
@@ -126,3 +131,85 @@ def _pymatgen_species(
         pymatgen_species.append(dict(zip(chemical_symbols, concentration)))
 
     return pymatgen_species
+
+
+def from_pymatgen(pmg_structure: Structure) -> StructureResourceAttributes:
+    """Convert a pymatgen `Structure` (3D) into an OPTIMADE `StructureResourceAttributes` model.
+
+    Parameters:
+        pmg_structure: The pymatgen `Structure` to convert.
+
+    Returns:
+        An OPTIMADE `StructureResourceAttributes` model, which can be converted to a raw Python
+            dictionary with `.dict()` or to JSON with `.json()`.
+
+    """
+
+    if not isinstance(pmg_structure, Structure):
+        raise RuntimeError(
+            f"Cannot convert type {type(pmg_structure)} into an OPTIMADE `StructureResourceAttributes` model."
+        )
+
+    attributes = {}
+    attributes["cartesian_site_positions"] = pmg_structure.lattice.get_cartesian_coords(
+        pmg_structure.frac_coords
+    ).tolist()
+    attributes["lattice_vectors"] = pmg_structure.lattice.matrix.tolist()
+    attributes["species_at_sites"] = [_.symbol for _ in pmg_structure.species]
+    attributes["species"] = [
+        {"name": _.symbol, "chemical_symbols": [_.symbol], "concentration": [1]}
+        for _ in set(pmg_structure.composition.elements)
+    ]
+    attributes["dimension_types"] = [1, 1, 1]
+    attributes["nperiodic_dimensions"] = 3
```

Comment:
Could you not do something like this to get the info about the periodic boundaries? 
```suggestion
    attributes["dimension_types"] = [int(_) for _ in pmg_structure.lattice.pbc]
    attributes["nperiodic_dimensions"] = sum(attributes["dimension_types"])
```

Reply:
Thanks, didn't know about `lattice.pbc` (just assumed all `Structure` were 3D).

Reply:
Looks like this needed a pymatgen update, which I've just done in #1294